tests/zquantum/core/estimation/estimation_test.py

```diff
@@ -685,11 +760,87 @@ def test_calculate_exact_expectation_values(
+            expectation_values_list, target_expectations, estimation_tasks
+        ):
+            for cov, corr, target_corr, target_cov in zip(
+                expectation_values.estimator_covariances,
+                expectation_values.correlations,
+                target.correlations,
+                target.estimator_covariances,
+            ):
+                assert len(corr) == len(task.operator.terms)
+                assert len(corr) == len(task.operator.terms)
+
+                np.testing.assert_array_almost_equal(corr, target_corr, decimal=2)
+                np.testing.assert_array_almost_equal(cov, target_cov, decimal=2)
+
+    @pytest.mark.parametrize(
+        "estimation_tasks,target_expectations",
+        TEST_CASES_NONEIGENSTATES,
+    )
+    def test_correlation_and_covariance_when_averaging_for_non_eigenstates(
+        self, simulator, estimation_tasks, target_expectations
+    ):
+
+        expectation_values_list = estimate_expectation_values_by_averaging(
+            simulator, estimation_tasks
+        )
+        for expectation_values, target, task in zip(
+            expectation_values_list, target_expectations, estimation_tasks
+        ):
+            for cov, corr, target_corr, target_cov in zip(
+                expectation_values.estimator_covariances,
+                expectation_values.correlations,
+                target.correlations,
+                target.estimator_covariances,
+            ):
+                assert len(corr) == len(task.operator.terms)
+                assert len(cov) == len(task.operator.terms)
+
+                np.testing.assert_array_almost_equal(corr, target_corr, decimal=1)
+                # All covariances should be close to 0 when number_of_shots is high.
+                np.testing.assert_array_almost_equal(target_cov, cov, decimal=2)
+
+    @pytest.mark.parametrize(
+        "mock_estimation_tasks, target_expectations",
+        TEST_CASES_NONEIGENSTATES_WITH_LOW_NUMBER_OF_SHOTS,
+    )
+    def test_covariance_when_averaging_for_non_eigenstates_and_number_of_shots_is_low(
+        self, target_expectations, mock_estimation_tasks
+    ):
+
+        expectation_values_list = estimate_expectation_values_by_averaging(
+            MockBackendForTestingCovariancewhenNumberOfShotsIsLow(),
+            mock_estimation_tasks,
+        )
+        for expectation_values, target in zip(
+            expectation_values_list, target_expectations
+        ):
+            for cov, corr, target_corr, target_cov in zip(
+                expectation_values.estimator_covariances,
+                expectation_values.correlations,
+                target.correlations,
+                target.estimator_covariances,
+            ):
+
+                np.testing.assert_array_almost_equal(corr, target_corr, decimal=2)
+                # All covariances should be close to 0 when number_of_shots is high.
+                np.testing.assert_array_almost_equal(cov, target_cov, decimal=2)
```

Comment:
The only covariance close to zero I've found is the test example 4  (the matrix with `0.05` on every element). Do other test examples also fit this assertion?

Reply:
Oops! that was left over from a copy/paste I did. thanks!